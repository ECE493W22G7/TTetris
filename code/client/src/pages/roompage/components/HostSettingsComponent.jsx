import "./gameSettingsComponent.css";
import { Button, Card } from "react-bootstrap"

// Reference for wifi button: https://fontawesomeicons.com/bootstrap/icons/wifi-off

const HostSettingsComponent = (props) => {
    const { isHost, gameSpeed, onGameSpeedChanged, onStartButtonClicked } = props;
    return (
        <Card className="game-settings-component">
            <div className="settings-content">
                <span className="game-title font-italic text-center">
                    Host Settings
                </span>
                <div className="host-input">
                    <div className="input-group">
                            <span className="input-group-text" id="inputGroup-sizing-default">
                                Set Speed:
                            </span>
                            <input
                                className="form-control set-speed"
                                type="range"
                                min="1"
                                max="15"
                                onChange={onGameSpeedChanged}
                                value={gameSpeed}
                                disabled={!isHost}
                                step="1"
                            ></input>
                    </div>
                    <Button
                        onClick={onStartButtonClicked}
                        disabled={!isHost}
                        className="start-button"
                        role="button"
                    >
                        Start Game
                    </Button>
                </div>
            </div>
        </Card>
    );
};

export default HostSettingsComponent;
