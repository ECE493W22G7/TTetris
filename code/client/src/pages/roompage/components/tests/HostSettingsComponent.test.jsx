import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import HostSettingsComponent from '../HostSettingsComponent';
import MockedSocket from 'socket.io-mock';
import socketIOClient from 'socket.io-client';
import { toMatchDiffSnapshot } from 'snapshot-diff';
const {act} = renderer;

test('renders correctly', () => {
    const tree = renderer
        .create(<HostSettingsComponent socket={null} roomID={"34223"}/>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});