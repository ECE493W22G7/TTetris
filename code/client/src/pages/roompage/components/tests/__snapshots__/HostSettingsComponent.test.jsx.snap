// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="game-settings-component card"
>
  <div
    className="settings-content"
  >
    <span
      className="game-title font-italic text-center"
    >
      Host Settings
    </span>
    <div
      className="host-input"
    >
      <div
        className="input-group"
      >
        <span
          className="input-group-text"
          id="inputGroup-sizing-default"
        >
          Set Speed:
        </span>
        <input
          className="form-control set-speed"
          disabled={true}
          max="15"
          min="1"
          step="1"
          type="range"
        />
      </div>
      <button
        className="start-button btn btn-primary"
        disabled={true}
        role="button"
        type="button"
      >
        Start Game
      </button>
    </div>
  </div>
</div>
`;
